### PR TITLE
max, min, email and url

### PR DESCRIPTION
**Pull request by paixaop**
Hi I added validators for min, max, email and url
min (>=) and max (<=) work like above and bellow, but are inclusive of the given number.

email test for RFC822 address
url tests for valid URLs

hope you can merge these back in.

### CODE REVIEW

**Comment by faridnsh** [comment]
Use regex.test instead of exec.

File: lib/should.js [hunk 1]
```diff
@@ -371,6 +371,38 @@ Assertion.prototype = {
...
+   * @api public
+   */
+
+  min: function(n, desc){
+    this.assert(
+        this.obj >= n
+      , function(){ return 'expected ' + this.inspect + ' to be above or equal ' + n + (desc ? " | " + desc : "") }
+      , function(){ return 'expected ' + this.inspect + ' to be below or equal ' + n + (desc ? " | " + desc : "") });
+    return this;
+  },
```
Comment (tj) [inline lib/should.js:405]: we already have above / below methods

**Comment by btd** [comment]
As Tj already said, min/max already available but with another names. I do not think specific regexps should be in should.js - at least while it is testing library.

File: lib/should.js (hunk 1 above)
Comment (paixaop) [inline lib/should.js:405]: Above/Bellow only test for bigger/smaller than, not equality. How would you test if a variable has a value bigger or equal using above? If we assume the value n is an integer you could test n-1, but what if n is a float?
Example: How can we test if a variable is bigger or equal to 9.8 using above?
Same thing for bellow. I agree that if above/bellow also tested equality min/max would not be needed.

**Comment by paixaop** [comment]
BTD I agree with you, not having specific regexps in a testing library, but some regexps like email validity testing are very common, and are really easy to get wrong. The likely outcome of each user writing their own regexp to test the validity of an email address is a lot of bugs.

File: lib/should.js (hunk 1 above)
Comment (btd) [inline lib/should.js:405]: For floating pointer numbers you should **never** use ==/=== to compare.

``` js
 ➜ node
> 0.3 == 0.1 + 0.2
false
```

**Comment by btd** [comment]
You can every time create a plugin for should.js. Just extend `Assertion.prototype` and require where you need it.

**Pull request closed without merging**